lib/ui/home/tabs/watches_tab.dart

```diff
@@ -1,96 +1,342 @@
-  @override
-  State<StatefulWidget> createState() => new _MyWatchesTabState();
-}
+class MyWatchesTab extends HookWidget {
+  final Color _disconnectedColor = Color.fromRGBO(255, 255, 255, 0.5);
+  final Color _connectedColor = Color.fromARGB(255, 0, 255, 170);
+  final Color _connectedBrColor = Color.fromARGB(255, 0, 169, 130);
 
-class _MyWatchesTabState extends State<MyWatchesTab> {
-  List<PebbleDevice> _pebbles = [] ;
-
-  @override
-  void initState() {
-    super.initState();
-  }
+  void getCurrentWatchStatus() {}
+  final ConnectionControl connectionControl = ConnectionControl();
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-        appBar: AppBar(
-          title: Text("My Watches"),
-          leading: BackButton(),
-        ),
-        body: ListView(children: <Widget>[
-          Column(
-              children: _pebbles
-                  .map((e) => InkWell(
-                child: Container(
+    final connectionState = useProvider(connectionStateProvider.state);
+    final defaultWatch = useProvider(defaultWatchProvider);
+    final pairedStorage = useProvider(pairedStorageProvider);
+    final allWatches = useProvider(pairedStorageProvider.state);
+
+    List<PebbleDevice> allWatchesList =
+        allWatches.map((e) => e.device).toList();
+    if (defaultWatch != null && connectionState.isConnected == true) {
+      //Hide the default watch if we're connected to it. We don't need to see it twice!
+      allWatchesList.remove(defaultWatch);
+    }
+
+    List<PebbleDevice> defaultWatchList;
+    if (defaultWatch != null) {
+      defaultWatchList = [defaultWatch];
+    } else {
+      defaultWatchList = [];
+    }
+
+    String statusText;
+    bool isConnected;
+    if (connectionState.isConnecting == true) {
+      statusText = "Connecting...";
+      isConnected = false;
+    } else if (connectionState.isConnected == true) {
+      statusText = "Connected!";
+      isConnected = true;
+    } else {
+      statusText = "Disconnected";
+      isConnected = false;
+    }
+
+    Color _getBrStatusColor(PebbleDevice device) {
+      if (connectionState.currentWatchAddress == device.address)
+        return _connectedBrColor;
+      else
+        return _disconnectedColor;
+    }
+
+    Color _getStatusColor(PebbleDevice device) {
+      if (connectionState.currentWatchAddress == device.address)
+        return _connectedColor;
+      else
+        return _disconnectedColor;
+    }
+
+    void _onDisconnectPressed(bool inSettings) {
+      connectionControl.disconnect();
+      if (inSettings) Navigator.pop(context);
+    }
+
+    void _onConnectPressed(PebbleDevice device, inSettings) {
+      NumberWrapper addressWrapper = NumberWrapper();
+      addressWrapper.value = device.address;
+      connectionControl.connectToWatch(addressWrapper);
+      if (inSettings) Navigator.pop(context);
+    }
+
+    void _onForgetPressed(PebbleDevice device) {
+      pairedStorage.unregister(device.address);
```

Comment:
This would probably also need to unpair the device from the system's bluetooth settings. Any chance you can create extra issue for that just so we don't forget it?

Reply:
Opened #70